### ISSUE TITLE
Doesn't work properly on recent Onyx firmwares

### ISSUE THREAD

**Issue by CrazyCoder** [calin-darie/gentle-glow-onyx-boox#20]
Onyx Leaf 2 and Onyx Page with 3.3.2 firmware are affected.

When I change the sliders of any preset, it switches automatically to the "Onyx slider" radio button, making it impossible to create and save presets with your own values.

Is there any quick hack to fix this? Thanks.

**Comment by CrazyCoder** [comment]
Thanks for the fix! I tested it on Onyx Page and Onyx Leaf 2 with 3.3.2 firmware and it now works correctly.

Doesn't work on Onyx Palma that has 3.4 firmware and the new controls for the front lights: https://help.boox.com/hc/en-us/articles/10701137102100-Front-Lights#title2.

**Comment by calin-darie** [comment]
🤦‍♀️sorry apparently there was still an upgrade I hadn't performed. 

Can you please check again on the Tab Ultra and Palma?

In case the latest update still doesn't work, I opened an issue with Onyx - not sure when I'll get a reply.

**Comment by CrazyCoder** [comment]
I'll try when the 4.1 release is available. Or do you want me to build it myself?

**Comment by CrazyCoder** [comment]
Just built 4.1 version and it still doesn't work at all on Onyx Palma.

**Comment by calin-darie** [comment]
Sorry, wasn't relying on you to build it, I had released it on Play Store last evening, before commenting here.

Just realized you were waiting for the release on Github, you were not going through the Play Store.

**Comment by calin-darie** [comment]
So we're stuck waiting for an update from Onyx I guess. I couldn't find the new API in the SDK referenced by Gentle Glow. Maybe it's a different library altogether?

**Comment by calin-darie** [comment]
https://github.com/calin-darie/gentle-glow-onyx-boox/releases/tag/v4.3rc1

Hi! If you have some time, please give this a spin. And if it works, please look at the debug logs so I can listen only for relevant system events.

**Comment by CrazyCoder** [comment]
Getting this:

![image](https://github.com/calin-darie/gentle-glow-onyx-boox/assets/122834/cbd771dc-fdd3-4d16-a5d0-73ae6dfa3998)

**Comment by calin-darie** [comment]
https://github.com/calin-darie/gentle-glow-onyx-boox/releases/tag/v4.3rc2

I'm pretty confident that this will.. at least break in a different way

**Comment by CrazyCoder** [comment]
This works! Night/Day/Sunset presets work fine, but the Dawn preset disables the frontlight checkbox and I have to enable it back.

Here is the log, the last part is enabling Dawn:

```
09-22 20:37:32.983  1366  1366 D OnyxPowerManager:  ========== cancel alarm:onyx_standby_timeout ==========
09-22 20:37:32.997   766   766 I SDM     : update_to_display[1/0] -- marker[1123] waveform_mode = 255, update_mode = 0, Rect[254 42 1191 740], flags = 0
09-22 20:37:33.026  1741  1741 I ThreePointConsumer: three point gesture invalid
09-22 20:37:33.029  1366  9168 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_br/brightness, value: 20
09-22 20:37:33.048  1366  1386 V WindowManager: ActivityRecord{5ce6082 u0 com.onyx/.phone.main.ui.PhoneMainActivity t371} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:33.048  1366  1386 V WindowManager: Task{25a3b01 #371 visible=true type=standard mode=fullscreen translucent=false A=1000:com.onyx.main U=0 StackId=371 sz=1} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:33.048  1366  1386 V WindowManager: App is requesting an orientation, return 1 for display id=0
09-22 20:37:33.048  1366  1386 V WindowManager: DefaultTaskDisplayArea@32893213 is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:33.056  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:33.060  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:33.065   766   766 I SDM     : update_to_display[1/0] -- marker[1124] waveform_mode = 255, update_mode = 0, Rect[101 42 1496 740], flags = 0
09-22 20:37:33.074  1366  1386 V WindowManager: ActivityRecord{5ce6082 u0 com.onyx/.phone.main.ui.PhoneMainActivity t371} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:33.074  1366  1386 V WindowManager: Task{25a3b01 #371 visible=true type=standard mode=fullscreen translucent=false A=1000:com.onyx.main U=0 StackId=371 sz=1} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:33.074  1366  1386 V WindowManager: App is requesting an orientation, return 1 for display id=0
09-22 20:37:33.074  1366  1386 V WindowManager: DefaultTaskDisplayArea@32893213 is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:33.078   805   805 E SurfaceFlinger: [com.onyx.darie.calin.gentleglowonyxboox/com.onyx.darie.calin.gentleglowonyxboox.FrontLightWarmthBrightnessDialog#0] rejecting buffer: bufWidth=740, bufHeight=1496, front.active_legacy.{w=740, h=1191}
09-22 20:37:33.081   766   766 I SDM     : update_to_display[1/0] -- marker[1125] waveform_mode = 255, update_mode = 0, Rect[254 42 1191 740], flags = 0
09-22 20:37:33.083  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_brightness_state_key
09-22 20:37:33.087  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_brightness_state_key
09-22 20:37:33.099   766   766 I SDM     : update_to_display[1/0] -- marker[1126] waveform_mode = 255, update_mode = 0, Rect[254 42 1394 740], flags = 0
09-22 20:37:33.114   766   766 I SDM     : update_to_display[1/0] -- marker[1127] waveform_mode = 255, update_mode = 0, Rect[0 0 1648 824], flags = 0
09-22 20:37:33.130   766   766 I SDM     : update_to_display[1/0] -- marker[1128] waveform_mode = 255, update_mode = 0, Rect[0 0 1648 824], flags = 0
09-22 20:37:34.794  1741  1741 I ThreePointConsumer: three point gesture invalid
09-22 20:37:34.818  1366  9171 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_br/brightness, value: 3
09-22 20:37:34.826  1366  9171 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_ct/brightness, value: 20
09-22 20:37:34.829   766   766 I SDM     : update_to_display[1/0] -- marker[1129] waveform_mode = 255, update_mode = 0, Rect[446 526 60 153], flags = 0
09-22 20:37:34.837  1366  1386 V WindowManager: ActivityRecord{5ce6082 u0 com.onyx/.phone.main.ui.PhoneMainActivity t371} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:34.837  1366  1386 V WindowManager: Task{25a3b01 #371 visible=true type=standard mode=fullscreen translucent=false A=1000:com.onyx.main U=0 StackId=371 sz=1} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:34.837  1366  1386 V WindowManager: App is requesting an orientation, return 1 for display id=0
09-22 20:37:34.837  1366  1386 V WindowManager: DefaultTaskDisplayArea@32893213 is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:34.845  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:34.847   766   766 I SDM     : update_to_display[1/0] -- marker[1130] waveform_mode = 255, update_mode = 0, Rect[253 42 1193 740], flags = 0
09-22 20:37:34.850  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:34.854  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:34.855  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:34.856  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_temperature_state_key
09-22 20:37:34.859  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_temperature_state_key
09-22 20:37:34.865   766   766 I SDM     : update_to_display[1/0] -- marker[1131] waveform_mode = 255, update_mode = 0, Rect[253 0 1193 782], flags = 0
09-22 20:37:34.881   766   766 I SDM     : update_to_display[1/0] -- marker[1132] waveform_mode = 255, update_mode = 0, Rect[0 0 1648 824], flags = 0
09-22 20:37:34.897   766   766 I SDM     : update_to_display[1/0] -- marker[1133] waveform_mode = 255, update_mode = 0, Rect[445 245 928 434], flags = 0
09-22 20:37:36.095  1741  1741 I ThreePointConsumer: three point gesture invalid
09-22 20:37:36.118  1366  9173 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_br/brightness, value: 16
09-22 20:37:36.128  1366  9173 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_ct/brightness, value: 7
09-22 20:37:36.130   766   766 I SDM     : update_to_display[1/0] -- marker[1134] waveform_mode = 255, update_mode = 0, Rect[537 554 60 125], flags = 0
09-22 20:37:36.134  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:36.136  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:36.139  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:36.140  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:36.146   766   766 I SDM     : update_to_display[1/0] -- marker[1135] waveform_mode = 255, update_mode = 0, Rect[445 271 928 408], flags = 0
09-22 20:37:36.163   766   766 I SDM     : update_to_display[1/0] -- marker[1136] waveform_mode = 255, update_mode = 0, Rect[703 245 380 324], flags = 0
09-22 20:37:36.197   766   766 I SDM     : update_to_display[1/0] -- marker[1137] waveform_mode = 255, update_mode = 0, Rect[537 554 60 125], flags = 0
09-22 20:37:37.199  1741  1741 I ThreePointConsumer: three point gesture invalid
09-22 20:37:37.217  1366  9175 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_br/brightness, value: 8
09-22 20:37:37.227  1366  9175 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_ct/brightness, value: 19
09-22 20:37:37.230   766   766 I SDM     : update_to_display[1/0] -- marker[1138] waveform_mode = 255, update_mode = 0, Rect[537 271 60 178], flags = 0
09-22 20:37:37.233  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:37.236  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:37.239  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:37.240  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:37.246   766   766 I SDM     : update_to_display[1/0] -- marker[1139] waveform_mode = 255, update_mode = 0, Rect[445 271 928 408], flags = 0
09-22 20:37:37.262   766   766 I SDM     : update_to_display[1/0] -- marker[1140] waveform_mode = 255, update_mode = 0, Rect[703 245 380 324], flags = 0
09-22 20:37:37.297   766   766 I SDM     : update_to_display[1/0] -- marker[1141] waveform_mode = 255, update_mode = 0, Rect[537 271 60 178], flags = 0
09-22 20:37:38.501  1741  1741 I ThreePointConsumer: three point gesture invalid
09-22 20:37:38.517  1366  9177 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_br/brightness, value: 0
09-22 20:37:38.530   766   766 I SDM     : update_to_display[1/0] -- marker[1142] waveform_mode = 255, update_mode = 0, Rect[445 271 60 155], flags = 0
09-22 20:37:38.546   766   766 I SDM     : update_to_display[1/0] -- marker[1143] waveform_mode = 255, update_mode = 0, Rect[445 271 928 408], flags = 0
09-22 20:37:38.596   766   766 I SDM     : update_to_display[1/0] -- marker[1144] waveform_mode = 255, update_mode = 0, Rect[445 271 60 155], flags = 0
09-22 20:37:39.070  1366  9177 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_ct/brightness, value: 0
09-22 20:37:39.070  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:39.074  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_brightness
09-22 20:37:39.077  1366  9177 I CTMController: ###setLightValue, path: /sys/class/backlight/onyx_bl_br/brightness, value: 0
09-22 20:37:39.100  1366  2656 V WindowManager: ActivityRecord{5ce6082 u0 com.onyx/.phone.main.ui.PhoneMainActivity t371} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:39.100  1366  2656 V WindowManager: Task{25a3b01 #371 visible=true type=standard mode=fullscreen translucent=false A=1000:com.onyx.main U=0 StackId=371 sz=1} is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:39.100  1366  2656 V WindowManager: App is requesting an orientation, return 1 for display id=0
09-22 20:37:39.100  1366  2656 V WindowManager: DefaultTaskDisplayArea@32893213 is requesting orientation 1 (SCREEN_ORIENTATION_PORTRAIT)
09-22 20:37:39.107  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_brightness_state_key
09-22 20:37:39.111  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_brightness_state_key
09-22 20:37:39.113   766   766 I SDM     : update_to_display[1/0] -- marker[1145] waveform_mode = 255, update_mode = 0, Rect[254 42 1191 740], flags = 0
09-22 20:37:39.113  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:39.115  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/screen_ctm_temperature
09-22 20:37:39.118  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_temperature_state_key
09-22 20:37:39.120  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_temperature_state_key
09-22 20:37:39.122  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_brightness_state_key
09-22 20:37:39.122  8982  8982 D ContentObserver: >> content changed for URI content://settings/system/ctm_brightness_state_key
09-22 20:37:39.130   766   766 I SDM     : update_to_display[1/0] -- marker[1146] waveform_mode = 255, update_mode = 0, Rect[254 42 1193 740], flags = 0
09-22 20:37:39.146   766   766 I SDM     : update_to_display[1/0] -- marker[1147] waveform_mode = 255, update_mode = 0, Rect[254 42 1191 740], flags = 0
09-22 20:37:39.163   766   766 I SDM     : update_to_display[1/0] -- marker[1148] waveform_mode = 255, update_mode = 0, Rect[0 0 1648 824], flags = 0
```

**Comment by calin-darie** [comment]
Thank you so much for all the support!

This issue is fixed in  v4.3 825522a2288a6c34bb45cc4ab8a41c6220d498aa
